--- ui.h
+++ ui.h
@@ -3,6 +3,7 @@
 #include "game.h"
 #include <SDL.h>
 #include <string>
+#include <unordered_set>
 #include <vector>
 
 namespace UI {
@@ -37,8 +38,13 @@
 };
 
 struct VisualNovelState {
     bool active = false;
-    std::string currentText = "Are you interested in recent stockings, Master?";
+    bool tutorialCompleted = false;
+    int tutorialStep = 0;
+    float textProgress = 0.0f;
+    std::string targetText = "";
+    std::string currentText = "";
+    std::unordered_set<int> visitedSystems;
+    int arrowTarget = 0;
 };
 
 struct WindowState {
